lib/webauthn/attestation_statement/base.rb

```diff
@@ -147,6 +150,32 @@ def raw_subject_key_identifier
         OpenSSL::ASN1.decode(ext_value.value).value
       end
 
+      def valid_signature?(authenticator_data, clien_data_hash)
+        public_key = attestation_certificate&.public_key || authenticator_data.credential.public_key_object
```

Comment:
This fallback to the credential public key only makes sense only for when attestation type is Self, which is only possible for `packed` format. We should not allow signature verification to continue if certificate public is not there for Basic or AttCA which is the case for other formats.

Reply:
Ah ok, sometimes this part of the puzzle gets a bit confusing to me. Thanks for catching it! I'll fix it.